app/date-select/controller.js

```diff
@@ -0,0 +1,34 @@
+const FormWizardController = require('../../common/controllers/form-wizard')
+const { dateRegex } = require('../../common/helpers/url')
+
+const DateRegExp = new RegExp('/' + dateRegex)
+
+class DateSelectController extends FormWizardController {
+  get(req, res, next) {
+    if (req.query.referrer) {
+      req.sessionModel.reset()
+      req.sessionModel.set('referrer', req.query.referrer)
+      return res.redirect(req.form.options.fullPath)
+    }
+
+    const referrer = req.sessionModel.get('referrer')
+
+    if (!referrer) {
+      return res.redirect('/')
+    }
+
+    res.locals.cancelUrl = referrer
+    res.locals.backLink = referrer
+
+    super.get(req, res, next)
+  }
+
+  successHandler(req, res, next) {
+    const { date_select: dateSelect, referrer } = req.sessionModel.toJSON()
+    req.sessionModel.reset()
```

Comment:
Should we also kill the `journeyModel`?

Reply:
that would lead to csrf issues if someone had double-tabbed. So why bother?

Reply:
The CSRF is no longer tied to the journey. That was the work I had to do previously around extracting it to the user’s global session. 

Before that work the FormWizard actually stored the CSRF token against the session so killing the session would have creates a CSRF issue when using multiple tabs. 

The reason for suggesting cleaning it up is that we won’t be left with old journeys on the global session that way.